framework/src/executioners/Executioner.C

```diff
@@ -35,120 +38,6 @@ validParams<Executioner>()
-  params.addParam<std::vector<std::string>>("splitting",
-                                            "Top-level splitting defining a "
-                                            "hierarchical decomposition into "
-                                            "subsystems to help the solver.");
-
-  std::set<std::string> line_searches = {"contact", "default", "none", "basic"};
-#ifdef LIBMESH_HAVE_PETSC
-  std::set<std::string> petsc_line_searches = Moose::PetscSupport::getPetscValidLineSearches();
-  line_searches.insert(petsc_line_searches.begin(), petsc_line_searches.end());
-#endif // LIBMESH_HAVE_PETSC
-  std::string line_search_string = Moose::stringify(line_searches, " ");
-  MooseEnum line_search(line_search_string, "default");
-  std::string addtl_doc_str(" (Note: none = basic)");
-  params.addParam<MooseEnum>(
-      "line_search", line_search, "Specifies the line search type" + addtl_doc_str);
-  MooseEnum line_search_package("petsc moose", "petsc");
-  params.addParam<MooseEnum>("line_search_package",
-                             line_search_package,
-                             "The solver package to use to conduct the line-search");
-  params.addParam<unsigned>("contact_line_search_allowed_lambda_cuts",
-                            2,
-                            "The number of times lambda is allowed to be cut in half in the "
-                            "contact line search. We recommend this number be roughly bounded by 0 "
-                            "<= allowed_lambda_cuts <= 3");
-  params.addParam<Real>("contact_line_search_ltol",
-                        "The linear relative tolerance to be used while the contact state is "
-                        "changing between non-linear iterations. We recommend that this tolerance "
-                        "be looser than the standard linear tolerance");
-
-// Default Solver Behavior
-#ifdef LIBMESH_HAVE_PETSC
-  params += Moose::PetscSupport::getPetscValidParams();
-#endif // LIBMESH_HAVE_PETSC
-  params.addParam<Real>("l_tol", 1.0e-5, "Linear Tolerance");
-  params.addParam<Real>("l_abs_step_tol", -1, "Linear Absolute Step Tolerance");
-  params.addParam<unsigned int>("l_max_its", 10000, "Max Linear Iterations");
-  params.addParam<unsigned int>("nl_max_its", 50, "Max Nonlinear Iterations");
-  params.addParam<unsigned int>("nl_max_funcs", 10000, "Max Nonlinear solver function evaluations");
-  params.addParam<Real>("nl_abs_tol", 1.0e-50, "Nonlinear Absolute Tolerance");
-  params.addParam<Real>("nl_rel_tol", 1.0e-8, "Nonlinear Relative Tolerance");
-  params.addParam<Real>("nl_abs_step_tol", 1.0e-50, "Nonlinear Absolute step Tolerance");
-  params.addParam<Real>("nl_rel_step_tol", 1.0e-50, "Nonlinear Relative step Tolerance");
-  params.addParam<bool>(
-      "snesmf_reuse_base",
-      true,
-      "Specifies whether or not to reuse the base vector for matrix-free calculation");
-  params.addParam<bool>("no_fe_reinit", false, "Specifies whether or not to reinitialize FEs");
```

Comment:
Note: I deleted this parameter because it seems no codes are using this parameter. It could be a left-over of previous refactorings.